gqcpy/src/QCMethod/HF/UHF/UHFSCFEnvironment_bindings.cpp

```diff
@@ -88,21 +88,49 @@ void bindUHFSCFEnvironment(py::module& module) {
             "coefficient_matrices_beta",
             &GQCP::UHFSCFEnvironment<double>::coefficient_matrices_beta)
 
-        .def_readonly(
+        .def(
```

Comment:
Could you explain why you changed from `.def_readonly` to `.def`?

Reply:
When using `def_readonly()` I get an error while compiling gqcpy:

`no matching member function for call to 'def_readonly'`
`candidate template ignored: could not match 'const D C::*' against '(lambda at /Users/xeno/Documents/GQCP/gqcpy/src/QCMethod/HF/UHF/UHFSCFEnvironment_bindings.cpp:93:13)'`

I wanted to ask you about this, since I wasn't sure about changing the `def_readonly()` bindings to regular `def()` ones myself. But I was first trying to fix the convergence problems.

Reply:
I looked it up in the [documentation](https://pybind11.readthedocs.io/en/stable/classes.html#instance-and-static-fields). Did you perhaps look over this section? `.def_readonly` can only be used to bind a C++ class' _member_, which is not what we're trying to accomplish here. However, once the Python bindings contain classes for the appropriate return types, (probably in #677), it will be possible to re-write this using the previous `.def_readonly` implementation.

Also, in these density/Fock matrix bindings, you've written code to transform a `deque` to a `vector`. While I appreciate the effort of manually writing this loop, there are a couple of improvements that can be made:
- Lose the C-style loop, and rewrite it using `for (const auto& density_matrix : environment.density_matrices)`;
- Or, and this is preferred, use the C++ STL's [`std::transform`](https://en.cppreference.com/w/cpp/algorithm/transform) method to modify C++'s native types.

Reply:
I must indeed have looked over this section, because I remember spending quite some time looking up why `def_readonly()` did no longer work. But thanks for clarifying! I'll look into rewriting the loops a swell.